pandas/_libs/tslibs/timezones.pyx

```diff
@@ -275,3 +275,84 @@ cdef object get_dst_info(object tz):
         dst_cache[cache_key] = (trans, deltas, typ)
 
     return dst_cache[cache_key]
+
+
+def _infer_tzinfo(start, end):
+    def _infer(a, b):
+        tz = a.tzinfo
```

Comment:
this should be an outside function

Reply:
If we're going to bother changing it, might as well get rid of `_infer`.  It's only actually relevant in one case (of four)

Reply:
sure